diff --git a/engine/events/poisonstep.asm b/engine/events/poisonstep.asm
index 98a6e25..e3b45da 100644
--- a/engine/events/poisonstep.asm
+++ b/engine/events/poisonstep.asm
@@ -131,6 +131,9 @@ DoPoisonStep::
 	farcall GetPartyNick
 	ld hl, .PoisonFaintText
 	call PrintText
+	xor a
+	ld [wPokemonWithdrawDepositParameter], a
+	callfar RemoveMonFromPartyOrBox
 
 .mon_not_fainted
 	pop de
diff --git a/engine/events/whiteout.asm b/engine/events/whiteout.asm
index 0906187..a91f7c3 100644
--- a/engine/events/whiteout.asm
+++ b/engine/events/whiteout.asm
@@ -11,6 +11,7 @@ Script_Whiteout:
 	waitbutton
 	special FadeOutPalettes
 	pause 40
+	callasm OverworldGameOver
 	special HealParty
 	checkflag ENGINE_BUG_CONTEST_TIMER
 	iftrue .bug_contest
@@ -96,3 +97,7 @@ GetWhiteoutSpawn:
 .yes
 	ld [wDefaultSpawnpoint], a
 	ret
+
+OverworldGameOver:
+	farcall EmptyAllSRAMBanks
+	jp Reset
